sklearn/cluster/_agglomerative.py

```diff
@@ -74,7 +74,10 @@ def _fix_connectivity(X, connectivity, affinity):
             for j in range(i):
                 idx_j = np.where(labels == j)[0]
                 Xj = X[idx_j]
-                D = pairwise_distances(Xi, Xj, metric=affinity)
+                if affinity == "precomputed":
+                    D = X[np.ix_(idx_i, idx_j)]
```

Comment:
@TomDLT I see that https://github.com/scikit-learn/scikit-learn/pull/20531 raises an error for precomputed metrics. Does it make sense to complete the matrix like this?